openqabot/types/incident.py

```diff
@@ -144,3 +148,53 @@ def contains_package(self, requires: List[str]) -> bool:
                 if package.startswith(req) and package != "kernel-livepatch-tools":
                     return True
         return False
+
+    def has_failures(self, token) -> bool:
+        results = requests.get(
+            QEM_DASHBOARD + "/api/jobs/incident/" + str(self.id), headers=token
+        ).json()
+
+        failed_jobs = self.filter_failures(results)
+        if failed_jobs:
+            log.info("Found %s failed jobs for incident %s:", len(failed_jobs), self.id)
+            if log.isEnabledFor(LOG_DEBUG_LEVEL):
+                list(
+                    map(lambda job: self.log_debug_incident(job["job_id"]), failed_jobs)
+                )
+
+        if not results:
+            raise NoResultsError(
+                "Job setting %s not found for incident" % (str(self.id))
+            )
+
+        # we only need to check if there are any failed jobs
+        return any(failed_jobs)
+
+    def log_debug_incident(self, job_id):
+        log.debug(
+            "Job %s is not marked as acceptable for incident %s",
+            job_id,
+            self.id,
+        )
+
+    def filter_failures(self, results):
+        return [
+            res
+            for res in results
+            if res["status"] not in ("passed", "softfailed")
+            and not has_ignored_comment(res["job_id"], self.id)
+        ]
+
+
+@staticmethod
+def has_ignored_comment(job_id: int, inc: int):
+    ret = []
+    ret = requests.get(OPENQA_URL + "/api/v1/jobs/%s/comments" % job_id).json()
+    regex = re.compile(r"\@review\:acceptable_for\:incident_%s\:(.+)" % inc)
```

Comment:
Wait. This was already used somewhere. Did you just copy-paste that?

Reply:
@okurz can you suggest a better way? wasn't really a fan of this, but didn't want to give you a heart attack by adding another TODO

Reply:
Extract the existing code into something reusable and call it in both places

Reply:
The **TODO** says `remove almost duplicated code from Approver.is_job_marked_acceptable_for_incident`. I suggest you either kill the TODO or actually move the code if it's easy enough.